contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -31,7 +31,10 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
     constructor() public Lockable(address(0x0)) {}
 
     // TopUp heimdall fee
-    function topUpForFee(uint256 validatorId, uint256 heimdallFee) public {
+    function topUpForFee(uint256 validatorId, uint256 heimdallFee)
+        public
+        onlyWhenUnlocked
+    {
```

Comment:
@AyushKaul add a check if the validator is active or not, `require(isValidator(validator))`.

Reply:
done